core/src/main/java/org/jruby/Main.java

```diff
@@ -531,6 +537,26 @@ protected static int handleRaiseException(RaiseException rj) {
         }
     }
 
+    public static void printTruffleTimeMetric(String id) {
+        if (Options.TRUFFLE_METRICS_TIME.load()) {
+            final long millis = System.currentTimeMillis();
+            System.err.printf("%s %d.%03d%n", id, millis / 1000, millis % 1000);
+        }
+    }
+
+    private static void printTruffleMemoryMetric() {
+        if (Options.TRUFFLE_METRICS_MEMORY_USED_ON_EXIT.load()) {
+            System.gc();
```

Comment:
Ok, I now saw that you parse verbose:gc, nice. Ok, but you probably still want to call System.gc() multiple times to make sure it triggers. Then, I would assume that the verbose:gc output gives you a sum of all the objects that made it into GCed memory.

Looks good to me!